oap-server/server-alarm-plugin/src/main/java/org/apache/skywalking/oap/server/core/alarm/provider/RulesReader.java

```diff
@@ -203,4 +206,25 @@ private void readCompositeRuleConfig(Rules rules) {
             }
         });
     }
+
+    /**
+     * Read composite rule config into {@link DingtalkSettings}
+     */
+    private void readDingtalkConfig(Rules rules) {
```

Comment:
Please recheck the test case, seems this method is not covered. Also, please help on rechecking other `read*` methods in this class.

Reply:
Actually I have added test logic in RulesReaderTest.java  will check others. it may be related to Codecov detect logic. let me check